app/features/ai_resistant_assignment_generator/document_loaders.py

```diff
@@ -0,0 +1,373 @@
+from langchain_community.document_loaders import YoutubeLoader, PyPDFLoader, TextLoader, UnstructuredURLLoader, UnstructuredPowerPointLoader, Docx2txtLoader, UnstructuredExcelLoader, UnstructuredXMLLoader
+from langchain_community.document_loaders.csv_loader import CSVLoader
+from langchain_google_genai import ChatGoogleGenerativeAI
+from app.utils.allowed_file_extensions import FileType
+from app.api.error_utilities import FileHandlerError, ImageHandlerError
+from app.api.error_utilities import VideoTranscriptError
+from langchain_core.messages import HumanMessage
+from app.services.logger import setup_logger
+from langchain_text_splitters import RecursiveCharacterTextSplitter
+from langchain_core.documents import Document
+import os
+import tempfile
+import uuid
+import requests
+import gdown
+
```

Comment:
since we have been using document loader in multiple tools, shouldn't we move it to global util? @WilfredoAaronSosaRamosBoushtech